src/pql/schema.py

```diff
@@ -0,0 +1,152 @@
+step_extract_http = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "extract"},
+        "method": {"enum": ["http.get", "http.post"]},
+        "uri": {"type": "string"},
+    },
+    "required": ["step", "method", "uri"],
+    "if": {"properties": {"method": {"const": "http.post"}}},
+    "then": {"properties": {"params": {"type": "object"}}, "required": ["params"]},
+}
+
+step_extract_sql = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "extract"},
+        "method": {"const": "sql.postgres"},
+        "uri": {"type": "string"},
+        "query": {"type": "string"},
+    },
+    "required": ["step", "method", "uri", "query"],
+}
+
+step_extract_eth = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "extract"},
+        "method": {"enum": ["eth.balance", "eth.function"]},
+        "address": {"type": "string"},
+    },
+    "required": ["step", "method", "address"],
+    "if": {"properties": {"method": {"const": "eth.balance"}}},
+    "then": {
+        "properties": {
+            "params": {
+                "type": "object",
+                "properties": {
+                    "block": {"oneOf": [{"const": "latest"}, {"type": "integer"}]},
+                    "num_confirmations": {"type": "integer"},
+                },
+                "required": ["block"],
+            }
+        }
+    },
+    "else": {
+        "properties": {
+            "params": {
+                "type": "object",
+                "properties": {
+                    "function": {"type": "string"},
+                    "args": {"type": "array", "items": {"type": "string"}},
+                    "num_confirmations": {"type": "integer"},
+                    "block": {"oneOf": [{"const": "latest"}, {"type": "integer"}]},
+                },
+                "required": ["function", "args", "block"],
+            },
+        },
+        "required": ["params"],
+    },
+}
+
+step_traverse = {
+    "type": "object",
+    "properties": {
+        "method": {"const": "json"},
+        "params": {"type": "array", "items": {"type": "string"}},
+    },
+    "required": ["method", "params"],
+}
+
+step_get_index = {
+    "type": "object",
+    "properties": {"step": {"const": "get_index"}, "params": {"type": "integer"}},
+    "required": ["step", "params"],
+}
+
+step_math = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "math"},
+        "method": {"enum": ["mul", "add", "sub", "div"]},
+        "params": {"type": "number"},
+        "direction": {"const": {"reverse"}},
+    },
+    "required": ["step", "method", "params"],
+}
+
+step_custom = {
+    "type": "object",
+    "properties": {"step": {"enum": []}},
+    "required": ["step"],
+}
+
+step_query_sql = {
+    "type": "object",
+    "properties": {
+        "step": {"const": "query.sql"},
+        "method": {"enum": ["json", "list", None]},
```

Comment:
What does it mean for method to be `None`?

Reply:
This corresponds to the case in which we do not need to parse the data before we push it into the SQLite table as defined by [this](https://github.com/frisitano/paralink-node/blob/feature/pql_validation/src/pql/query_sql.py#L24) condition.